odra-vm/src/vm/odra_vm.rs

```diff
@@ -58,7 +58,7 @@ impl OdraVm {
     }
 
     pub fn call_contract(&self, address: Address, call_def: CallDef) -> Bytes {
-        self.prepare_call(address, &call_def.entry_point, call_def.amount);
+        self.prepare_call(address, call_def.clone());
```

Comment:
I would pass a reference and clone only as the last resort, so possibly in line 96